File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    var answer = 0;
    let arr = [];
    
    for(let i = 2; i <= n; i++) arr[i] = i;
    
    for(let i = 2; i <= n; i++){
        if(arr[i] === 0) continue;
        for(let j = i + i; j <= n; j+=i) arr[j] = 0;  
    }
    return arr.filter((e) => e).length;;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations

def check(num):
    if num <= 1: return False
    for j in range(2 , num):
        if num % j == 0: return False
    return True

def solution(numbers):
        
    unique_numbers = set()
    answer = 0
    for i in range(1, len(numbers) + 1):
        for perm in permutations(numbers, i):
            num_str = ''.join(perm)
            unique_numbers.add(int(num_str)) 
            
    for l in list(unique_numbers):
        print(l)
        if check(l): answer += 1
        
    return answer

